### ISSUE TITLE
Crash when applying aws_s3_bucket with empty cors allowed_origins

### ISSUE THREAD

**Issue by jc-asdf** [terraform-providers/terraform-provider-aws#4258]
### Terraform Version
Terraform v0.11.7
+ provider.aws v1.15.0

### Affected Resource(s)
aws_s3_bucket

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-east-1"
}

resource "aws_s3_bucket" "static" {
  bucket = "terraform-testing-asdfa"
  acl = "private"
  cors_rule {
    allowed_headers = ["*"]
    allowed_methods = ["PUT", "POST"]
    allowed_origins = [""]
    expose_headers  = ["ETag"]
    max_age_seconds = 3000
  }
}
```

### Debug Output
https://gist.github.com/jc-asdf/16ff36eadc5ab187971e31a2826d3c3a

### Panic Output
https://gist.github.com/jc-asdf/22d5189cf5ccb75d2d31d43b803b30cc.

### Expected Behavior
Either failure caused by AWS API due to missing "AllowedOrigins" parameter ("The XML you provided was not well-formed or did not validate against our published schema") or success, with Terraform creating an empty `<AllowedOrigins></AllowedOrigins>` tag

### Actual Behavior
Terraform crash.  The bucket gets created (with default CORS settings), but does not get saved in the terraform state.

### Steps to Reproduce
1. Provide an empty string (perhaps via a variable substitution mistake, like how I originally found this) as the only parameter for the `allowed_origins` parameter for the `cors_rule` block.

2. `terraform apply`

**Comment by bflad** [comment]
The fix for this has been merged via #4465 and will release with v1.18.0 of the AWS provider in about two days.

**Comment by bflad** [comment]
This has been released in [version 1.18.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1180-may-10-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
